wp-seo-main.php

```diff
@@ -27,6 +27,53 @@
 	define( 'WPSEO_CSSJS_SUFFIX', ( ( defined( 'SCRIPT_DEBUG' ) && true === SCRIPT_DEBUG ) ? '' : '.min' ) );
 }
 
+global $pagenow;
+if ( is_admin() && $pagenow === 'plugins.php' ) {
+	add_action( 'admin_init', 'yoast_seo_free_register_deactivate_listener' );
+	add_action( 'admin_init', 'yoast_seo_premium_register_deactivate_listener' );
+}
+
+/**
+ * Listener for Yoast SEO free. Disable Free when someone is trying to activate Premium.
+ */
+function yoast_seo_free_register_deactivate_listener() {
+	if ( ( $listener_plugin = filter_input( INPUT_GET, 'plugin' ) ) && $listener_plugin === 'wordpress-seo-premium/wp-seo-premium.php' ) {
```

Comment:
`filter_input( INPUT_GET, 'plugin' ) === 'wordpress-seo-premium/wp-seo-premium.php'` is better.